asciidoctorj-api/src/main/java/org/asciidoctor/ast/Document.java

```diff
@@ -75,4 +75,7 @@
      * @param state The state in which to put the sourcemap (true = on, false = off).
      */
     void setSourcemap(boolean state);
+
+
+    Catalog getCatalog();
```

Comment:
It would be great to have a Javadoc here stating that this will only return sth useful when called from within a Converter (and/or after Document.getContent() was called).

Reply:
Good idea, but this caveat only apply to footnotes.  
The catalog will soon also have `getRefs()` which will be available before `Document.getContent()` is called.
(I think I am right about the above, will verify/adjust javadocs as I add refs).